src/Components/FileExplorer/Component_FileExplorer.re

```diff
@@ -242,19 +243,29 @@ let update = (~configuration, msg, model) => {
 
     let model = {...model, treeView};
     switch (outmsg) {
-    | Expanded(node) => (
+    | Component_VimTree.Expanded(node) => (
         model,
         Effect(
           Effects.load(node.path, configuration, ~onComplete=newNode =>
             NodeLoaded(newNode)
           ),
         ),
       )
-    | Collapsed(_) => (model, Nothing)
-    | Selected(node) =>
+    | Component_VimTree.Collapsed(_) => (model, Nothing)
+    | Component_VimTree.Touched(node) =>
```

Comment:
I like the way you divided up the responsibility here - `Touched` vs `Selected`. I wonder if `Selected` should potentially preview as well - this would occur if the user selected a file via keyboard?

Reply:
I didn't find in a NerdTree any preview mode. So to match API it'll be just `Enter` to select.
To add preview capabilities we'd need to introduce some another keybinding lie `Shift+Enter` that can be mapped to `Touched`

Reply:
Ah, right, I think you mentioned this before - thanks for the details!